lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py

```diff
@@ -887,11 +887,12 @@ def test_missing_date_catch_warning(self, warning_list=None):
             predictor_cube, variance_cube, optimised_coeffs,
             self.coeff_names, predictor_of_mean_flag)
 
+        warning_msg = "Ensemble calibration not available"
```

Comment:
I think as well as modifying our test, so that we are only checking for the presence of a warning, rather than whether the warning is at a particular position within the warning list. we also want to remove the true on the line below that counts the number of warnings i.e.
```python
self.assertTrue(len(warning_list) == 1)
```
This will mean that the test won't fail, if the number of warnings increase e.g. due to changes in stacks. This affects a number of the other changes, where this line that counts the number of warnings can just be removed.